### ISSUE TITLE
Back-off pulling image \"openshift/origin-pod:v1.4+f4347b6

### ISSUE THREAD

**Issue by enlinxu** [openshift/origin#10346]
I modified the node.yaml.v1.j2 in openshift-ansible, to add the podManifestConfig, to deploy static pod while openshift starts.

However, when openshift started, I am seeing error when the static pod is starting:
Aug 10 18:05:16 vmt-osMaster origin-node[27759]: E0810 18:05:16.814470   27759 pod_workers.go:183] Error syncing pod c9891bf2bd2f76eb21c7124c9aa837ae, skipping: failed to "StartContainer" for "POD" with ImagePullBackOff: "Back-off pulling image \"openshift/origin-pod:v1.4+f4347b6\""
Aug 10 18:05:31 vmt-osMaster origin-node[27759]: I0810 18:05:31.812858   27759 docker_manager.go:1742] Need to restart pod infra container for "kubeturbo-10.10.173.79_default(c9891bf2bd2f76eb21c7124c9aa837ae)" because it is not found
Aug 10 18:05:31 vmt-osMaster origin-node[27759]: E0810 18:05:31.819822   27759 docker_manager.go:1962] Failed to create pod infra container: ImagePullBackOff; Skipping pod "kubeturbo-10.10.173.79_default(c9891bf2bd2f76eb21c7124c9aa837ae)": Back-off pulling image "openshift/origin-pod:v1.4+f4347b6"

As you can see above, It is complaining the during pulling the image, and it is trying to pull "openshift/origin-pod:v1.4+f4347b6" , which is not the image I specified in my yaml.

I wonder where is this "openshift/origin-pod:v1.4+f4347b6" coming from? Why it's not following my image?

I tried docker pull to pull my image, and I could pull it successfully.
##### Version

[provide output of the `openshift version` or `oc version` command]
[root@OSMaster ~]# oc version
oc v1.2.0
kubernetes v1.2.0-36-g4a3f9c5

**Comment by detiber** [comment]
@enlinxu can you provide a copy of your master config file and where you are pulling the openshift binaries from? 

In general, the pod image version tags are defaulted to the server version, but can be overridden by overriding the value of imageConfig.format in the master/node configs (set by the oreg_url ansible variable).

**Comment by enlinxu** [comment]
@detiber Thanks for your reply. Actually I am pulling from my own repo. Basically I used tito to build the rpms of origin, and package and upload to my repo for a test. Do I need to change anything if I am using customized repo?
Thanks

**Comment by detiber** [comment]
@enlinxu you would need to either override the oreg_url variable (or set imageConfig.format in the master/node configs) to use a version available in the upstream repos, or you would need to build and host the images related to your build.

**Comment by enlinxu** [comment]
@detiber Thanks again. So if I want to make sure my static is running after openshift-ansible playbook finishes, I need to add my own repo to the /etc/ansible/hosts as:
[OSEv3:vars]
ansible_ssh_user=root
deployment_type=origin
oreg_url=docker.io/my-repo/

Is that going to work? Or I have to override the imageConfig.format in master/node config after the playbook finishes

**Comment by sdodson** [comment]
@enlinxu the version of oc is different from the version of the master and node components you're running? 

You'd want something like this assuming you had docker.io/my-repo/origin-pod:v1.4 and all the others there too. The master/node binaries have the version encoded in them and will ask for the version that matches them.
oreg_url='docker.io/my-repo/origin-${component}:${version}'

**Comment by enlinxu** [comment]
Thanks @sdodson . So if I want to just test if the rest of the things is working, by hard code the version in imageConfig: of node and master config, which version I should use? And where can I get the version info?

**Comment by sdodson** [comment]
@enlinxu You can just use latest assuming that image tag is up to date whenever you build.

**Comment by enlinxu** [comment]
I am a bit confused about how the version is generated. If I am using tito to build the rpm from the upstream, then I am getting:
-rw-r--r--.  1 root root 42279280 Aug 12 12:04 origin-0.0.1-0.git.13099.c7de01e.el7.src.rpm

But if I do the tito build on my forked origin, in which I just added some logs for testing, I am getting:
-rw-r--r--.  1 root root 42317586 Aug 12 14:16 origin-0.0.2-1.git.3760.8ac166a.el7.src.rpm

the 0.0.2 is because I did the tito tag, but what about the difference of the version, i.e 13099.c7de01e v.s. 3760.8ac166a

Thanks a lot for all the help.
